### ISSUE TITLE
Instructions still valid? -- https://learn.microsoft.com/en-gb/sharepoint/per-machine-installation

### ISSUE THREAD

**Issue by gaiking-uk** [MicrosoftDocs/OfficeDocs-SharePoint#3690]
### Page content

On the page https://learn.microsoft.com/en-gb/sharepoint/per-machine-installation, it advises that...
> By default, the OneDrive sync app installs per-user

... and that this can be converted to a per-machine install by running ```OneDriveSetup.exe /allusers``` to perform an 'over-the-top' install of OneDrive (which installs OneDrive to the "Program Files" directory).

***

### However
- OneDrive in *installed* as part of Windows (at least in the case of Windows 10 and 11) and is already running when the user logs in the first time - i.e. neither the admin or user has ever run ```OneDriveSetup.exe```

- When I tried to follow the instructions provided, i.e. downloading and running the [**latest** version of OneDriveSetup.exe](https://go.microsoft.com/fwlink/?linkid=844652) with the ```/allusers``` option, I got an error saying that my existing/installed version of OneDrive was newer than the *latest* one available from Microsoft

![image](https://user-images.githubusercontent.com/37931190/223823003-72007a0d-cc3b-4516-84f2-87c66598ac66.png)

***

### Issue/Question

- Based on the above info, I wanted to confirm if the per-machine / over-the-top installation approach was still valid?
- Is it expected that the *latest* downloadable version of OneDriveSetup.exe was older than the version I had pre-installed by Windows?
- Please can the page be amended to clarify if this install process is still valid or not, and if so what additional steps are needed (e.g. if uninstalling OneDrive manually first, etc)

Thanks 👍🏼

---
#### Document details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: d03e432e-17b8-cdf8-5cd1-158f19d493e4
* Version Independent ID: 699feed2-7123-c54d-8e01-4be30466f3e4
* Content: [Install the sync app per-machine (Windows) - SharePoint in Microsoft 365](https://learn.microsoft.com/en-gb/sharepoint/per-machine-installation)
* Content Source: [SharePoint/SharePointOnline/per-machine-installation.md](https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/blob/public/SharePoint/SharePointOnline/per-machine-installation.md)
* Service: **one-drive**
* GitHub Login: @MikePlumleyMSFT
* Microsoft Alias: **mikeplum**

**Comment by HansBrender** [comment]
@gaiking-uk : The site, where you got the  "latest" version is not updated often enough. So the user has a newer version installed

**Comment by scanum** [comment]
We appreciate any feedback that improves the content in the Microsoft docs but we haven't had any response from you so we are going to close this issue.  
 
Please feel free to re-open this issue if there is a specific area of the docs that we can improve or make better. Thank you.

**Comment by gaiking-uk** [comment]
> @gaiking-uk : The site, where you got the "latest" version is not updated often enough. So the user has a newer version installed So which version do you have installed on the device and which ring (Insiders, Production, Deferred)?

I appreciate the term is somewhat vague, but note...
- That terminolgy comes from the page itself: 
![image](https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/assets/37931190/2d48f54d-4dfc-4b3e-800d-e16012ddf7fe)
- The link above points to: https://go.microsoft.com/fwlink/?linkid=844652
- This link ***directly initiates*** the download of the `OneDriveSetup.exe` program (rather than pointing to a download page) so I am unable to provide any other info

**Comment by HansBrender** [comment]
@gaiking-uk 
As I mentioned before, the site is more often not in sync . And you have a newer version installed. In my Blog I try to follow  and display [all versions.](https://hansbrender.com/versionen/) 

Why do you want tro change a newer version, which is installed on your device?
That is the error message you give us.

**Comment by gaiking-uk** [comment]
@HansBrender - I realise my replies are v. slow and intermittent however this issue feels a bit like an up-hill battle where the simple fix would have been to add a step to the deployment instructions along the lines of: "If you get an error advising that you have a newer version installed already, please manually uninstall before continuing" but instead I have to defend *_why_* I am trying to follow a Microsoft guide and move from per-user to per-machine setup.

The deployment instructions are still incorrect, or at least incomplete IMO but this issue has been closed now and so if Microsoft think that the below scenario is fine then I guess there is nothing to change...

***

#### DEPLOYMENT INSTRUCTIONS
1. Download `OneDriveSetup.exe`
2. Run `OneDriveSetup.exe /allusers` from a command prompt window

#### EXPECTED BEHAVIOUR
- `OneDriveSetup.exe` runs and converts the *per-user* installation to a *per-machine* installation

#### ACTUAL BEHAVIOUR
- `OneDriveSetup.exe` **FAILS BY DESIGN**
- User gets an error messaging saying a newer version is already installed
- OneDrive is not converted from a *per-user* installation to a *per-machine* installation

#### REASON FOR BEHAVIOUR
- The deployment will **ALWAYS FAIL** because:
- *BY DESIGN THERE WILL BE A NEWER VERSION* already installed, because the installed app gets more frequent updates than the installer, so the installer is always older than in the version on the system
- `OneDriveSetup.exe` is *SPECIFICALLY CODED TO ABORT* install if there is a newer version already installed, which there will be
- The deployment instructions make **NO MENTION** of the possible need to remove the existing *per-user* installation before being able to successfully run the *per-machine* installation.

***

> @HansBrender: Why do you want to change a newer version, which is installed on your device?

If you have read the page, it explains the reasons why users may want to do this quite clearly (I don't want to be rude, I  just mean that the reasons users will try and follow these steps are right there at the top of the page)...

"By default, the OneDrive sync app installs per-user, meaning that you'll need to install the app for each user on a machine. With the per-machine installation option, you'll only need to install the app once on a PC. This option is especially useful for computers with multiple users and for when you don't want executable files running from a user profile."
https://learn.microsoft.com/en-gb/sharepoint/per-machine-installation

**Comment by HansBrender** [comment]
There are 2 diffrent installtions possible. 
1) Install insider a Users directory
2) Install it as a per machine installation

2nd one is only inetering, if
a) you are running in a Citrix or Server environment
b) if you have a machine, where more than one  user  is working

OneDrive Version: 
Each day, the OneDrive client looks, if there is a new version. If yes , the app installed the new version

**You should**  (must) deinstall the old OneDrive Version first, but keep in mind, that you have to put all  synchronisations manually after installion of OneDrive

But again: Why do you want to install OneDrive as a machine.
It's the same program, and mit has NOTING to do with your issues

if you have an error with you current version: Open a ticket with microsoft

**Comment by gaiking-uk** [comment]
Hi Hans, thanks for getting back to me, please see my comments below...

> @HansBrender: You should (must) deinstall the old OneDrive Version first

**FINALLY!**... I am glad we are finally on the same page! 🎉... ***This is exactly what I have been saying since the beginning!***
- Yes, because of the way the setup app is coded it will always fail because a newer version is already installed
- **THE ISSUE IS:** the deployment instructions **DO NOT MENTION THIS**!
- The deployment instructions suggest to go from *per user* to *per machine*, all you need to do is simply re-run `OneDriveSetup.exe /allusers` over the top of your existing OneDrive, to convert it the the *per-machine* version
- There is nothing on the page that tells users that they need to uninstall OneDrive first
- The ***WHOLE*** point of raising this issue was to check if:
   1. `OneDriveSetup.exe` failing to install in this situation is expected -- ✅ Which it is
   2. If so, **update the deployment instructions** on https://learn.microsoft.com/en-gb/sharepoint/per-machine-installation to add `If OneDrive is already installed [which it will be, as Windows ships with OneDrive already installed], you need to manually uninstall OneDrive before installing the per-machine version` -- ❌ Still not done after 6 months

I really don't understand why it takes ***SO MUCH*** effort to get Microsoft to add 1 sentence in order to clarify some deployment instructions!

***

> @HansBrender: Why do you want to install OneDrive as a machine.

You keep asking this question, so for the record...
- I am doing this (well, I did this 6 months ago)† because I have multiple users on one machine and just prefer to have programs stored in a single location, rather than in each user's directory
- Again, this is pretty much exactly what is on the page: `This option is especially useful for computers with multiple users and for when you don't want executable files running from a user profile.`

I appreciate that for you, maybe the *why* was important, but to be asked this over and over just seemed like you were missing the point. Also, just to re-iterate that I ***completed the switch to per-machine install months ago***, since then all I have wanted to do is get a page updated!?

#### Imagine the following situation...

1. There is a Microsoft page on how to seamlessly go from the Windows Standard Build to the Windows Insider Beta...
2. You like seeing what features are upcoming and the page says you can do this by ***simply downloading and running the installer***.
3. You download and run the installer, but get an error saying you need to do a clean install of Windows to install the Windows Insider Beta.
4. You think that's a bit odd as the instructions make no mention of this at all, but you go ahead and do the fresh install to get the Windows Insider Beta.
5. You decide to try and be a good member of the community, and point out to Microsoft that the current page doesn't say anywhere that a fresh install is needed (which will likely surprise or confuse users) so they can improve the page by making this clear
6. Rather than getting a reply that goes "thanks for pointing that out Hans, we appreciate you taking the time to let us know and have added a note to the page to clarify this"... You instead spend 6 months arguing with Microsoft, where instead of just fixing the page, they point out that you need to do a fresh install and keep ask you "***why*** you want to install Windows Insider Beta!?"

***

I hope this clarifies why I found this issue so exhausting. **PLEASE** can the page just be updated, I don't think there is any way I can express the above more clearly, and I am getting frustrated that it has to take **HOURS** of my time to get Microsoft to spend **2 MINUTES** adding some text to a web page.

**Comment by gaiking-uk** [comment]
Still no changes to: https://learn.microsoft.com/en-gb/sharepoint/per-machine-installation
🤷🏼‍♂️ Give up

**Comment by HansBrender** [comment]
@gaiking-uk after 3 week ? I wait on a nother site since 2 years. Please be patient

**Comment by gaiking-uk** [comment]
I can be patient, but there was no reply to my last message so I did not even know ***IF*** the site will be updated, now that Microsoft finally understand the problem with the page.

I am happy to wait longer, but I will leave this to you and the Microsoft team to update now... I have spent many hours trying to explain the problem since I first reported it half a year ago.

It is up to Microsoft now to add 1 sentence to the page, between steps [1] and [2] to clarify that the existing version of OneDrive needs to be uninstalled before installing the `/allusers` version... Hopefully Microsoft can add 1 sentence in 2024 or 2025.

Thank you for your help on this.
